app/event_images/event_images.py

```diff
@@ -0,0 +1,133 @@
+    'gym': ['gym', 'workout', 'workouts'],
+    'haircut': ['haircut', 'hair'],
+    'halloween': ['halloween', 'helloween', "hallowe'en",
+                  'Allhalloween', "All Hallows' Eve", "All Saints' Eve"],
+    'hike': ['hiking', 'hike', 'hikes'],
+    'kayak': ['kayaking'],
+    'music': ['piano', 'singing', 'music Class', 'choir practice',
+              'flute', 'orchestra', 'oboe', 'clarinet', 'saxophone',
+              'cornett', 'trumpet', 'contrabass', 'cello', 'trombone',
+              'tuba', 'music ensemble', 'string quartett',
+              'guitar lesson', 'classical music', 'choir'],
+    'manicure': ['manicure', 'pedicure', 'manicures', 'pedicures'],
+    'massage': ['massage', 'back rub', 'backrub', 'massages'],
+    'pill': ['pills', 'medicines', 'medicine', 'drug', 'drugs'],
+    'pingpong': ['ping pong', 'table tennis', 'ping-pong', 'pingpong'],
+    'plan': ['plan week', 'plan quarter', 'plan day', 'plan vacation',
+             'week planning', 'vacation planning'],
+    'pokemon': ['pokemon'],
+    'read': ['reading', 'newspaper'],
+    'repair': ['fridge repair', 'handyman', 'electrician', 'DIY'],
+    'ran': ['jog', 'jogging', 'running', 'jogs', 'runs'],
+    'sail': ['sail', 'sailing', 'boat cruise', 'sailboat'],
+    'santa': ['Santa Claus', 'Father Christmas'],
+    'ski': ['skiing', 'ski', 'skis', 'Snowboarding', 'snowshoeing',
+            'snow shoe', 'snow boarding'],
+    'soccer': ['soccer'],
+    'swam': ['swim', 'swimming', 'swims'],
+    'tennis': ['tennis'],
+    'thanksgiving': ['thanksgiving'],
+    'wed': ['wedding', 'wedding eve', 'wedding-eve party', 'weddings'],
+    'christmas': ['christmas', 'xmas', 'x-mas'],
+    'yoga': ['yoga'],
+}
+
+
+def generate_flare_link_from_lemmatized_word(lemmatized_word: str) -> str:
+    """Generate a link to a flair by a given lemmatized word.
+
+    Args:
+        lemmatized_word (str): The lemmatized word.
+
+    Returns:
+        str: The suitable link.
+    """
+    return f'{FLAIRS_REL_PATH}\\{lemmatized_word}{FLAIRS_EXTENSION}'
+
+
+def remove_non_alphabet_chars(s: str) -> str:
+    """Remove non-alphabet chars from a given string
+
+    Args:
+        s (str): The string to remove the non-alphabet chars from.
+
+    Returns:
+        str: The string after the removal.
+    """
+    regex = re.compile('[^a-zA-Z]')
+    return regex.sub('', s)
+
+
+def get_key(val: str) -> Optional[str]:
+    """Search the key of a given value in IMAGES_RELATED_WORDS_MAP dictionary.
+
+    Args:
+        val (str): The value to search its key.
+
+    Returns:
+        str: The value's key in IMAGES_RELATED_WORDS_MAP dictionary..
+    """
+    shrunken = remove_non_alphabet_chars(val).lower()
+    for key, values in IMAGES_RELATED_WORDS_MAP.items():
+        shrunken_list = [remove_non_alphabet_chars(v).lower() for v in values]
+        if shrunken in shrunken_list:
+            return key
+
+
+def search_token_in_related_words(token: str) -> Optional[str]:
```

Comment:
We might want to cache this function

Reply:
I'd go with something bigger. 4,096 words will include about 95% of the usages in English, and this is pretty reasonable space to allocate in memory :)